tests/subsys/fs/littlefs/src/test_lfs_basic.c

```diff
@@ -485,6 +485,68 @@ static int check_large(void)
 	return TC_PASS;
 }
 
+static int check_ram_disk(void)
+{
+	struct fs_mount_t *mp = &testfs_ram_disk_mnt;
+	struct fs_statvfs stat;
+	struct fs_littlefs *littlefs_config = (struct fs_littlefs *)mp->fs_data;
+
+	zassert_equal(testfs_lfs_wipe_ram_disk(), TC_PASS,
+		      "clear partition failed");
+
+	TC_PRINT("program size: %zu, cache size: %zu\n",
+		littlefs_config->cfg.prog_size, littlefs_config->cfg.cache_size);
+
+	zassert_equal(fs_mount(mp), 0,
+		      "ram disk mount failed");
+
+	zassert_equal(fs_statvfs(mp->mnt_point, &stat), 0,
+		      "statvfs failed");
+
+	TC_PRINT("%s: bsize %lu ; frsize %lu ; blocks %lu ; bfree %lu\n",
+		 mp->mnt_point,
+		 stat.f_bsize, stat.f_frsize, stat.f_blocks, stat.f_bfree);
+	zassert_equal(stat.f_bsize, 16,
+		      "bsize fail");
+	zassert_equal(stat.f_frsize, 4096,
+		      "frsize fail");
+	zassert_equal(stat.f_blocks, 16,
+		      "blocks fail");
+	zassert_equal(stat.f_bfree, stat.f_blocks - 2U,
+		      "bfree fail");
+
+	zassert_equal(fs_unmount(mp), 0,
+		      "ram disk unmount failed");
+
+	return TC_PASS;
+}
+
+static int check_ram_disk_and_large(void)
```

Comment:
@Laczen Usage of a normal flash device and something different (in this case a in memory storage) could look like this.

Reply:
@benediktibk, nice work. What would happen (during mount) if storage_dev has no meaning ?

A proposal could be that if a high value is assigned to storage_dev (e.g. 0x8000) flash is not used and also the mount procedure could be changed to reflect this.

Another proposal could be that a change to littlefs_mount() is required - this seems reasonable as probably different backends require different checks for mounting.

Reply:
Thanks.
You had a good point there, I forgot to check the result of device_services->open. As of know the result of an invalid value for storage_dev depends on the behaviour of the implementation of device_services->open. In the case of flash this will result in an error, as flash_area_open will return an error.

Honestly, I did not get what you are aiming for with this special range of values for storage_dev. Why would we want to have a different mount procedure, besides the already possible adaption via device_services?

Reply:
> You had a good point there, I forgot to check the result of device_services->open. As of know the result of an invalid value for storage_dev depends on the behaviour of the implementation of device_services->open. In the case of flash this will result in an error, as flash_area_open will return an error.
>
 Could you modify the test so that for the RAM_DISK the value of storage_dev is not set.

> Honestly, I did not get what you are aiming for with this special range of values for storage_dev. Why would we want to have a different mount procedure, besides the already possible adaption via device_services?

I overlooked that you also modified the mount procedure. The basic idea should be OK. However it is not OK to check for 0 and if it is set it to the flash method. There should be some verification of device_services that all routines are available, not asign them. The asignment when 0 is OK for the "configuration variables" in little_fs but not for the routines. You might get a mix of flash/other backend routines.

I think it is possible to remove any configuration option just by using macro's:  the "GP" macro is allready there (I would rename it to `FS_LITTLEFS_DECLARE_CUSTOM_CONFIG_CUSTOM_OPS`, we need the following macro's that will call the "GP" macro:
a. `FS_LITTLEFS_DECLARE_DEFAULT_CONFIG_CUSTOM_OPS` that calls the GP macro with read_size, prog_size, cache_size, lookahead_size all 0,
b. `FS_LITTLEFS_DECLARE_CUSTOM_CONFIG` that calls the GP macro with read_size, ... custom values and sets the device_services to the flash services, and adds the flash api routines,
c. `FS_LITTLEFS_DECLARE_DEFAULT_CONFIG` that calls the macro under b with read_size, ... set to 0.

Reply:
done